test/parallel/test-fs-promises-readfile-empty.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+const common = require('../common');
+
+const assert = require('assert');
+const { promises: fs } = require('fs');
+const fixtures = require('../common/fixtures');
+
+const fn = fixtures.path('empty.txt');
+
+common.crashOnUnhandledRejection();
+
+fs.readFile(fn)
+  .then(assert.ok);
+
+fs.readFile(fn, 'utf8')
+  .then(assert.strictEqual.bind(this, ''));
```

Comment:
If you don't want to fix the encoding bug simply replacing the assertions with `assert.ok` should be fine. But the fix should probably be pretty simple (if there is an encoding, return a `''` instead of `Buffer.alloc(0)`) so I'd suggest to include it.